### PR TITLE
Wrap the code to make a good usaged of Select2 with AJAX

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#41617](https://central.tri.be/issues/41617)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$current_attr' | [Main.php:1759](https://github.com/moderntribe/the-events-calendar/blob/194531bb1d78b5eabac2b644fb46e145f387ba87/src/Tribe/Main.php#L1759)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$current_attr' | [Main.php:1783](https://github.com/moderntribe/the-events-calendar/blob/194531bb1d78b5eabac2b644fb46e145f387ba87/src/Tribe/Main.php#L1783)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Ajax/Select2.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	 * Response to Select2 to search Venues
+	 * This will produce a HTTP response with a JSON in the Select2 format
+	 *
+	 * @return HTTP wp_send_json
+	 */
+	public function search_venues() {
+		$response = (object) array(
+			'items' => array(),
+		);
+
```
Comment (borkweb) [inline src/Tribe/Ajax/Select2.php]: Should we bail here if `$_GET['q']` is empty?

Comment (bordoni) [inline src/Tribe/Ajax/Select2.php]: It never will be but, we can make sure it doesn't show a notice.

**Comment by bordoni** [comment]
Just updated, we want to be able to have results without the search that's why we don't bail.

**Comment by borkweb** [comment]
Sounds good! :cactus:

**Pull request merged**